test/org/javarosa/core/model/instance/TreeReferenceParentTest.java

```diff
@@ -0,0 +1,52 @@
+package org.javarosa.core.model.instance;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.javarosa.core.model.instance.TestHelpers.buildRef;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TreeReferenceParentTest {
+
+    @Parameterized.Parameter(value = 0)
+    public String testCase;
+
+    @Parameterized.Parameter(value = 1)
+    public String tr;
+
+    @Parameterized.Parameter(value = 2)
+    public String base;
+
+    @Parameterized.Parameter(value = 3)
+    public String expectedRef;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+            {"Parenting absolute refs doesn't change them", "/foo", "/bar", "/foo"},
+            {"Parenting to an empty ref doesn't change them", "../foo", "", "../foo"},
+            {"foo.parent(bar) gives bar/foo", "foo", "bar", "bar/foo"},
+            {"foo.parent(../bar) gives ../bar/foo", "foo", "../bar", "../bar/foo"},
+            // TODO review this. Shouldn't this resolve to "bar/foo"?
```

Comment:
Did you base these off the latest master? I think that's the bug that was fixed in another recent PR.

Reply:
I've rebased the PR. Let's see if that solved this.

Reply:
It looks like there's still a problem here, right? Would be good to file an issue because it should not be `null`.

Reply:
OK, I've created #362